src/main/java/com/google/api/codegen/util/ProtoParser.java

```diff
@@ -378,14 +391,15 @@ public boolean isHttpGetMethod(Method method) {
 
   /** The hostname for this service (e.g. "foo.googleapis.com"). */
   public String getServiceAddress(Interface service) {
-    return getProtoExtension(service, AnnotationsProto.defaultHost);
+    return getProtoExtension(service, ClientProto.defaultHost);
   }
 
   /** The OAuth scopes for this service (e.g. "https://cloud.google.com/auth/cloud-platform"). */
   public List<String> getAuthScopes(Interface service) {
-    OAuth oAuth = getProtoExtension(service, AnnotationsProto.oauth);
+    String oAuth = getProtoExtension(service, ClientProto.oauthScopes);
     if (oAuth != null) {
-      return ImmutableList.copyOf(oAuth.getScopesList());
+      return ImmutableList.copyOf(
+          Arrays.stream(oAuth.split(",")).map(String::trim).collect(Collectors.toList()));
```

Comment:
Do we need to check for isNullOrEmpty? And if not - can we remove the check on line 357?

Reply:
1. Oh yes.
2. For line 357, it needs the `isNullOrEmpty` check to catch an instance where some method signature is explicitly set to the non-null empty string `""`. Without the check, the method would return one of the method signatures being a List<String> containing the element `""`, when really we want to return an empty list.